lib/ecto/query/planner.ex

```diff
@@ -1784,7 +1784,8 @@ defmodule Ecto.Query.Planner do
   @all_exprs [with_cte: :with_ctes, distinct: :distinct, select: :select, from: :from, join: :joins,
               where: :wheres, group_by: :group_bys, having: :havings, windows: :windows,
               combination: :combinations, order_by: :order_bys, limit: :limit, offset: :offset]
-
+  @all_cte_exprs @all_exprs ++ [update: :updates]
```

Comment:
The order matters, because it is the orders of bindings, so I believe you will have to list all fields, with `update: :updates` coming immediately after `from: :from`.

Reply:
I've created a new attr list for CTE exprs in https://github.com/elixir-ecto/ecto/pull/4220/commits/879d58f52e0f11b40145cda9621add93532ca5c4 to place :update after :from